From patchwork Thu Aug  9 21:54:18 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Shier <pshier@google.com>
X-Patchwork-Id: 10562017
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 2790514C0
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  9 Aug 2018 21:54:26 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 173E32B365
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  9 Aug 2018 21:54:26 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 0B7ED2B907; Thu,  9 Aug 2018 21:54:26 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-15.5 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,
	USER_IN_DEF_DKIM_WL autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id B0C602B365
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  9 Aug 2018 21:54:25 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727333AbeHJAVL (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 9 Aug 2018 20:21:11 -0400
Received: from mail-it0-f73.google.com ([209.85.214.73]:43616 "EHLO
        mail-it0-f73.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727294AbeHJAVL (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 9 Aug 2018 20:21:11 -0400
Received: by mail-it0-f73.google.com with SMTP id d9-v6so33992itf.8
        for <kvm@vger.kernel.org>; Thu, 09 Aug 2018 14:54:23 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:message-id:mime-version:subject:from:to:cc;
        bh=SRwiNnOZKQExMj1HsRwTdkmWRmA+Tx21itfOKVn6jww=;
        b=Wxpc+jyXGsuEyUO8DIOy1twlzTkAGCdE0KSfMFVOvXmfq3TCkdQN5yUVO0PQ51Q8tq
         pe9Z5lDmr+SlM9Vv2Iy9d/sJC/JScJ5K4kNfv3vsppuZegX1GwkF46d1XoAnEP2zYrn5
         LFq9SQTYbi/sG1M/1qs5RbxpBxZ7zFOR9OCaUncKXvmTScg+6Ug+f8JSqKHrirUL+VD2
         FzWsqkclYsLVrM98nXx7Ry99Q0FDI+6gpXJ4i/T1cgjSBAx6qax/peH6fuZLFaHujnpA
         wCl+iXxxpKkjVFXukspcjiSqu/XxeAPAASAQX7gKKIWqVkj+JpWU3MSIdIcyd7FnoHUM
         v0Yw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:message-id:mime-version:subject:from:to:cc;
        bh=SRwiNnOZKQExMj1HsRwTdkmWRmA+Tx21itfOKVn6jww=;
        b=BKtbHBcj05BUmq5ueUZLEAj1pORjkOWoA6nv1a7Q8XBXIwINO5SYtMmUY78joCEoKJ
         zg55Pak79MhMLGv39NMsc/e248XcH4SuxSIfaRmOvNt8gWtO6ETA3OrqcTRQS60/I7h9
         pAfmYw4QYEhXRrN2LiJ5/3ta/UCwEG9pUuK6smIrPVF3jNdeczx4y36Ec7DtgjpsDE0c
         onx2jdhfOHe01bfiJ0Q6p/gx3V5fI90OoqJx53EXGsHmiHQjmPH5nmsrodMogmpCgsak
         gmGDd8uj8kE8WaaOn2XurplFco+ujj6MnKDi2PqFDpypyJGLJ5FmVLcLGuCc4q57D36E
         NeYQ==
X-Gm-Message-State: AOUpUlF9IM1pwKPsttPafkl4tQf+IkcFAQogACg46YnxL1BWXdo52IVr
        DtpQpGbzTDrjpqiuIR03fh6zMMpHW2ORPUSVEbKaHbdZdUavVd2HylaCULZcNoKTOMWOxEWydjx
        1+CIfPE9LDkOBjOnNPvZRXXdXNFyL0uRYY7A9r0NwqHIXtZ4LVPuib4y/4w==
X-Google-Smtp-Source: 
 AA+uWPwzBo/+VtiAzJCqoifyiyyCuvSxdGlRH73hQmqJZqVqTYC/FSFyzVwpv13uixD1SCtCjHPedlpBPqE=
X-Received: by 2002:a24:5d15:: with SMTP id w21-v6mr14706ita.45.1533851663111;
 Thu, 09 Aug 2018 14:54:23 -0700 (PDT)
Date: Thu,  9 Aug 2018 14:54:18 -0700
Message-Id: <20180809215418.5519-1-pshier@google.com>
Mime-Version: 1.0
X-Mailer: git-send-email 2.18.0.597.ga71716f1ad-goog
Subject: [PATCH 1/2] kvm: expose KVM_CAP_ENABLE_CAP in KVM_CHECK_EXTENSION
From: Peter Shier <pshier@google.com>
To: kvm@vger.kernel.org
Cc: Peter Shier <pshier@google.com>
Content-Type: text/plain; charset="UTF-8"
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

KVM_ENABLE_CAP support was added to x86 in
commit 5c919412fe61 ("kvm/x86: Hyper-V synthetic interrupt controller")
but it was not exposed via KVM_CHECK_EXTENSION.

Signed-off-by: Peter Shier <pshier@google.com>
Reviewed-by: Jim Mattson <jmattson@google.com>
---
 arch/x86/kvm/x86.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 3c83711c0ebe1..3af45954a957f 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -2888,6 +2888,7 @@ int kvm_vm_ioctl_check_extension(struct kvm *kvm, long ext)
 	case KVM_CAP_PCI_SEGMENT:
 	case KVM_CAP_DEBUGREGS:
 	case KVM_CAP_X86_ROBUST_SINGLESTEP:
+	case KVM_CAP_ENABLE_CAP:
 	case KVM_CAP_XSAVE:
 	case KVM_CAP_ASYNC_PF:
 	case KVM_CAP_GET_TSC_KHZ:
